### ISSUE TITLE
configure error: checking for EIGEN... CMake Errors

### ISSUE THREAD

**Issue by richelbilderbeek** [cooljeanius/gcta#3]
Dear GCTA maintainer,

Here I report two `./configure` errors and one success in the hope of building GCTA.

I tried to build GCTA using the instructions in INSTALL, after installing EIGEN. I use Ubuntu 20.04 LTS (Focal Fossa), so installing EIGEN goes like this: 

```
sudo apt install libeigen3-dev 
```

I have CMake version 3.16.3:

```
$ cmake --version
cmake version 3.16.3
```

When running `./configure` I got this error (full output at the end of this Issue):

```
checking for EIGEN... CMake Error: Problem processing arguments. Aborting.

CMake Error: Problem processing arguments. Aborting.

no
```

When I updated CMake to 3.20.3, the `./configure` error changes to:

```
checking for EIGEN... CMake Error: Run 'cmake --help' for all supported options.
CMake Error: Run 'cmake --help' for all supported options.
no
```

On a [GitHub Actions script](https://github.com/richelbilderbeek/gcta/runs/2543013156?check_suite_focus=true#step:6:10), 
which uses CMake 3.20.2, this step does pass.

I think this is a bug in `configure` (one in three times it does
pass), maybe it is a CMake-version-specific syntax instead?

To be helpful, I copied the relevant lines of `configure` here:

```
failed=false
{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for EIGEN" >&5
printf %s "checking for EIGEN... " >&6; }
if test -z "${EIGEN_CXXFLAGS}"; then
    EIGEN_CXXFLAGS=`$CMAKE_BINARY --find-package "-DNAME=EIGEN" "-DCOMPILER_ID=GNU" "-DLANGUAGE=CXX" -DMODE=COMPILE ` || failed=true
fi
if test -z "${EIGEN_LIBS}"; then
    EIGEN_LIBS=`$CMAKE_BINARY --find-package "-DNAME=EIGEN" "-DCOMPILER_ID=GNU" "-DLANGUAGE=CXX" -DMODE=LINK ` || failed=true
fi
```

How can I configure GCTA? Or, how to modify `configure` to do so?

BTW, I also made [the Pull Request to add GitHub Actions](https://github.com/cooljeanius/gcta/pull/2).

Thanks and cheers, Richel Bilderbeek

## Full `./configure output`

```
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking which of GNU rx or gawk's regex is wanted... regex
checking for GNU regex in libc... yes
checking dependency style of gcc... none
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... none
checking for rm... /usr/bin/rm
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for printf in -lc... yes
checking for gzread in -lz... yes
checking for main in -leigen... no
checking for main in -lcrt0.o... no
checking if g++ supports namespace std... yes
checking whether complex math functions are in std::... yes
checking whether the compiler implements namespaces... yes
checking whether the compiler has numeric_limits<T>... yes
checking whether the compiler has complex math functions... yes
checking whether the compiler has more complex math functions... no
checking whether the compiler has complex<T>... yes
checking whether the compiler has std::vector::at (std::size_t)... yes
checking for ISO C++ 98 include files... yes
checking for ISO C++ TR1 include files... yes
checking for gawk... (cached) gawk
checking for GNU make... make
checking for cmake... /usr/bin/cmake
checking for EIGEN... no
checking for stdio.h... yes
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for size_t... yes
checking for atoi... yes
checking for abs... yes
checking for pow... no
checking for reverse... no
checking for sqrt... no
checking for seed_str... no
checking for stable_sort... no
checking for str_strm... no
checking for time... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating Makefile_orig
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
```

**Comment by richelbilderbeek** [comment]
Actually, one workaround is to comment out (i.e. add `#` before each line) these lines of `configure`:

```
if $failed; then
    unset EIGEN_CXXFLAGS
    unset EIGEN_LIBS

    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
printf "%s\n" "no" >&6; }

else
    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
printf "%s\n" "yes" >&6; }

fi
```

**Comment by cooljeanius** [comment]
Hi, I'm not actually a GCTA maintainer, I'm just some random person who made a fork of it in the process of trying to package it... I guess I can look into this anyways, though...

> Actually, one workaround is to comment out (i.e. add `#` before each line) these lines of `configure`:
> 
> ```
> if $failed; then
>     unset EIGEN_CXXFLAGS
>     unset EIGEN_LIBS
> 
>     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
> printf "%s\n" "no" >&6; }
> 
> else
>     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
> printf "%s\n" "yes" >&6; }
> 
> fi
> ```

`configure` is generated from `configure.ac` by autoconf, so this change should be done there instead.

**Comment by cooljeanius** [comment]
ok this should be fixed by c18d2906c7bfc49c8f5eeee14f953c9661f2cf96
